code/processes/idb.q

```diff
@@ -0,0 +1,45 @@
+/-default parameters
+\d .idb
+
+/- loads the idb and the sym file for given partition
+loaddb:{[pt]
+    ptdir: ` sv  .idb.savedir,(`$string[pt]),`;
+    if[() ~ key ptdir;.lg.e[`load;"missing intraday db folder: ",string ptdir];.idb.setupsuccess:0b];
+    @[system; "l ", 1_string ptdir; {.lg.e[`load;"failed to reload db from folder: ",string[y]," error: ",x];.idb.setupsuccess:0b}[;ptdir]];
+    symfilepath: ` sv .idb.savedir,`sym;
+    @[load; symfilepath; {.lg.e[`load;"failed to reload sym file: ",x];.idb.setupsuccess:0b}];
+ };
+
+/- reloads the db. Called by wdb process midday/eod.
+intradayreload:{[pt]
+    /- first check if IDB is properly setup, if not, try to setup again
+    if[not .idb.setupsuccess;setup[.idb.savedir;pt]];
+    loaddb[pt];
```

Comment:
if `setup` was called, this has already been done I think - is there harm in redoing?

Reply:
Reworked as well. This will first check a change in sym file size/partition count then do a reload accordingly.